src/pymovements/dataset/dataset_definition.py

```diff
@@ -222,7 +223,17 @@ def to_dict(self, exclude_private: bool = True) -> dict[str, Any]:
                 if key.startswith('_'):
                     del data[key]
 
-        data['experiment'] = data['experiment'].to_dict()
+        # Delete fields that evaluate to False (False, None, [], {})
+        if exclude_none:
+            if not bool(self.experiment):
+                del data['experiment']
+            else:
+                data['experiment'] = data['experiment'].to_dict()
+            for key, value in list(data.items()):
+                if not value:
```

Comment:
This should probably be constricted to non-boolean (and non-integer) fields, otherwise they will be also excluded if they're false (or zero). write instead: 
```python
if not isinstance(value, (bool, int, float)) and not value:
```